### ISSUE TITLE
Legend shapes disappear with transparent fill

### ISSUE THREAD

**Issue by domoritz** [vega/vega#946]
See https://github.com/vega/vega-lite/issues/2719. 

```json
{
  "$schema": "https://vega.github.io/schema/vega/v3.0.json",
  "data": [
    {
      "name": "source_0",
      "url": "data/cars.json",
      "format": {
        "type": "json",
        "parse": {
          "Horsepower": "number",
          "Miles_per_Gallon": "number"
        }
      }
    }
  ],
  "scales": [
    {
      "name": "shape",
      "type": "ordinal",
      "domain": {"data": "source_0","field": "Cylinders","sort": true},
      "range": "symbol"
    }
  ],
  "legends": [
    {
      "shape": "shape",
      "title": "Cylinders"
    },
    {
      "shape": "shape",
      "title": "Cylinders",
      "encode": {
        "symbols": {"update": {"fill": {"value": "transparent"}}}
      }
    }
  ]
}
```

![image](https://user-images.githubusercontent.com/589034/28749714-a1b4b354-7485-11e7-9a9a-65a35293d2b9.png)

**Comment by kanitw** [comment]
Im sure this is due to the fix for https://github.com/vega/vega-parser/issues/46

**Comment by jheer** [comment]
@kanitw is right. I think we need to revert vega/vega-parser#46 and only suppress the default stroke when the fill color is bound to a scale at the top-level of the legend (`{"fill": "fillScale", ...}`. We currently suppress the default stroke when a custom fill encoding is applied as well, but this issue clearly shows that this was a poor choice...

**Comment by domoritz** [comment]
Okay, in that case @kanitw, please close https://github.com/vega/vega-lite/pull/2721.

**Comment by jowens** [comment]
When this is fixed, please post "fixed in <a particular vega release>" so I can upgrade my headers accordingly? Thanks! (Or if there's an automatic way to mark issues-fixed-in-release-N in github that will post here or elsewhere, that'd be awesome.)

**Comment by jheer** [comment]
Yes, will do.

**Comment by jheer** [comment]
This is fixed by vega/vega-parser@6e5564cc9cf35fa33b541c76ac39fc4613f206cb, and will be included in the Vega release (version 3.0.0-rc5).

**Comment by jowens** [comment]
I confirm 3.0.0-rc5 fixes this issue.
